pyexchange/uniswapv2_analytics.py

```diff
@@ -266,24 +305,39 @@ def get_trades(self, pair: str, page_number: int = 1) -> List[Trade]:
         pair_address = self.get_pair_address(base_token.address, quote_token.address)
 
         mint_events = self.get_our_mint_txs(pair_address)
+        burn_events = self.get_our_burn_txs(pair_address)
         last_mint_event = mint_events[0]
+        last_burn_event = mint_events[0]
 
         our_liquidity_balance = Wad.from_number(last_mint_event['liquidity'])
         current_liquidity = self.get_current_liquidity(pair_address)
 
         last_mint_timestamp = int(last_mint_event['timestamp'])
+        last_burn_timestamp = int(last_burn_event['timestamp'])
         two_days_ago_unix = int(time.time() - (49 * 60 * 60))
+        mints_in_last_two_days = list(filter(lambda mint: int(mint['timestamp']) > two_days_ago_unix, mint_events))
+
+        current_time = int(time.time())
 
-        if current_liquidity == Wad.from_number(0):
+        if len(mint_events) == 0:
             return trades_list
-        elif last_mint_timestamp > two_days_ago_unix and len(mint_events) == 1:
-            start_timestamp = last_mint_timestamp
-        else:
+        elif current_liquidity != Wad.from_number(0) and len(mints_in_last_two_days) >= 1:
+            start_timestamp = mints_in_last_two_days[-1]['timestamp']
+            end_timestamp = current_time
+        elif current_liquidity != Wad.from_number(0) and len(mints_in_last_two_days) == 0:
             start_timestamp = two_days_ago_unix
+            end_timestamp = current_time
+        elif current_liquidity == Wad.from_number(0) and len(mints_in_last_two_days) >= 1:
+            start_timestamp = mints_in_last_two_days[-1]['timestamp']
+            end_timestamp = last_burn_timestamp
+        elif current_liquidity == Wad.from_number(0) and len(mints_in_last_two_days) == 0:
```

Comment:
Seems a bit redundant; the `else` below could cover this case.  Unless you just wanted it here for setting a breakpoint.

Reply:
Yeah that was the idea, but that would be much less confusing for future readers. I will update it accordingly.